metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -1606,6 +1614,17 @@ class MetalsLanguageServer(
         quickConnectToBuildServer().asJavaObject
       case ServerCommands.DisconnectBuildServer() =>
         disconnectOldBuildServer().asJavaObject
+      case ServerCommands.DecodeFile() =>
+        val argsMaybe = Option(params.getArguments())
+        argsMaybe.flatMap(_.asScala.headOption) match {
+          case Some(arg: JsonPrimitive) =>
+            val uri = arg.getAsString()
```

Comment:
Out of curiosity, what made you choose to go with a `uri` here? From a pure client implementation perspective I find this a bit more confusing that having this just be a json object that we pass in. We also don't do this `uri` format in any other place as far as I'm aware so we're sort of breaking pattern here. I'm not against this, more just curious why you chose this and if there is an advantage that I'm not seeing.

Reply:
You mean instead of passing URI as a String, you'd want something like this?...
```
{
  "uri" : "someURI"
}
```

I can change the format - the URI is the only data you have to hand when implementing a Virtual Document so I just passed it on

Reply:
> the URI is the only data you have to hand when implementing a Virtual Document so I just passed it on

Ahh ok, this explains it then. I didn't know this. I'd say don't change it then. I was thinking more like the example you showed in the other comment down below:

```{
  "file": "file:///somePath/someFile.java.javap.verbose",
  "decoder": "javap",
  "verbose": "true"
}
```

However, if that makes it more difficult for VS Code and virtual documents, then for sure don't change it. 

I typically default to do what is best for VS Code since it has the highest numbers, but do it in a way that still works in other clients. This will still work fine in other clients, was more just curious about the choice. 👍🏼 I'd say leave this as is.